src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,1441 @@
+    }
+
+    if (visualizer.preset == Preset.Primes) {
+        visualizer.backgroundColor = BLACK
+        visualizer.property1 = Property.Prime
+        visualizer.property1Visualization = PropertyVisualization.Color
+        visualizer.property1MainColor = RED
+    } else if (visualizer.preset == Preset.Abundant_Numbers) {
+        visualizer.backgroundColor = WHITE
+        visualizer.property1 = Property.Abundant
+        visualizer.property1Visualization = PropertyVisualization.Color
+        visualizer.property1MainColor = BLACK
+    } else if (visualizer.preset == Preset.Abundant_Numbers_And_Primes) {
+        visualizer.backgroundColor = WHITE
+        visualizer.property1 = Property.Abundant
+        visualizer.property1Visualization = PropertyVisualization.Color
+        visualizer.property1MainColor = BLACK
+        visualizer.property2 = Property.Prime
+        visualizer.property2Visualization = PropertyVisualization.Color
+        visualizer.property2MainColor = RED
+    } else if (visualizer.preset == Preset.Polygonal_Numbers) {
+        visualizer.backgroundColor = BLACK
+        visualizer.property1 = Property.Triangular_Number
+        visualizer.property1Visualization = PropertyVisualization.Color
+        visualizer.property1MainColor = RED
+        visualizer.property2 = Property.Square_Number
+        visualizer.property2Visualization = PropertyVisualization.Color
+        visualizer.property2MainColor = ORANGE
+        visualizer.property3 = Property.Pentagonal_Number
+        visualizer.property3Visualization = PropertyVisualization.Color
+        visualizer.property3MainColor = YELLOW
+        visualizer.property4 = Property.Hexagonal_Number
+        visualizer.property4Visualization = PropertyVisualization.Color
+        visualizer.property4MainColor = GREEN
+        visualizer.property5 = Property.Heptagonal_Number
+        visualizer.property5Visualization = PropertyVisualization.Color
+        visualizer.property5MainColor = BLUE
+        visualizer.property6 = Property.Octagonal_Number
+        visualizer.property6Visualization = PropertyVisualization.Color
+        visualizer.property6MainColor = PURPLE
+    }
+}
+
+function setOverridingSettings(visualizer: VisualizerGrid) {
+    if (visualizer.resetAndAugmentByOne) {
+        visualizer.pathType = PathType.Rows
+    }
+
+    if (
+        (visualizer.visualInfo == VisualInfo.Show_Numbers
+            || visualizer.visualInfo == VisualInfo.Show_Path)
+        && visualizer.amountOfNumbers > 400
+    ) {
+        visualizer.amountOfNumbers = 400
+    }
+
+    if (
+        visualizer.formula == Formula.TWO_TO_THE_N
+        && visualizer.amountOfNumbers > 1000
+    ) {
+        visualizer.amountOfNumbers = 1000
+    }
+}
+
+function getNumberFromFormula(
+    visualizer: VisualizerGrid,
+    currentSequenceIndex: number,
+    augmentForSequence: number
+) {
+    let sequenceElementAsNumber = Number(
+        visualizer.seq.getElement(currentSequenceIndex)
+    )
```

Comment:
There is a reason that Sequence instances return bigints -- many of the values of OEIS sequences are much too large to be represented by JavaScript's number type (which is a 64-bit IEEE floating-point value). So you should leave sequence values as bigints, and compute using that type to avoid loss of precision. Otherwise, you run into such weird behavior as "all sufficiently large integers are even" (really!). This will get somewhat easier when you move the transformations out to a Sequence class, so there will be less sequence computation done internally here. There may be some difficulties with checking some properties, especially as mathjs does not yet have bigint support (I am working on that), so if you can't implement them all fully within bigint, and are really forced within some of the checking to convert to number, then you must check for "out of range" conditions (just in those cases) and not visualize entries you can't properly compute on -- otherwise you will just be providing misinformation to the viewer.

Reply:
You said you "heard changing to bigints could be done later"... where did you hear that? We want to stick with bigint manipulation insofar as possible in Numberscope, because of the serious pitfalls of trying to compute with large integers in the number class. Please revert as many instances of sequence values from Number to bigint as possible, before this is merged. Thanks.

Reply:
I made this correction.

Reply:
Will look again once this is up-to-date.

Reply:
All right, bigint use seems to be consistent now, thanks!